File: c++/easy/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ret;
        if(nums.size() == 0)
            return ret;
        unordered_map<int, int> map;
        for(int i = 0; i < nums.size(); ++i)
        {
            int rest = target - nums[i];
            if(map.find(rest) != map.end())
            {
                int idx1 = i;
                int idx2 = map[rest];
                ret.push_back(idx1);
                ret.push_back(idx2);
                return ret;
            }
            else
                map[nums[i]] = i;
        }
    }
};

Conclusion:
easy! hashtable!
time complexity: O(n)
space complexity: O(n)

File: python/easy/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        ret = []
        if len(nums) == 0:
            return ret
        d = {}
        idx = 0
        for idx in xrange(0, len(nums)):
            rest = target - nums[idx]
            if rest in d:
                ret.append(idx)
                ret.append(d[rest])
                return ret
            d[nums[idx]] = idx
        
